### ISSUE TITLE
Support for Apple’s TSO patch?

### ISSUE THREAD

**Issue by eddyg** [lima-vm/lima#2865]
### Description

Wondering about the possibility of having templates for things like Ubuntu with [Apple’s TSO patch](https://developer.apple.com/documentation/virtualization/accelerating_the_performance_of_rosetta) pre-applied to improve performance of x86 workloads:

> x86 CPUs implement a stricter memory modern than ARM64 (TSO). For this reason, x86 emulation on baseline ARM64 systems requires very expensive memory model emulation. Having hardware that supports this natively is therefore very attractive. Such hardware, in fact, exists. This patch series adds support for userspace to identify when TSO is available and toggle it on, if supported.

[Apple tried](https://lwn.net/Articles/969480/) upstreaming these kernel patches, as did [Hector Martin of Asahi Linux](https://lore.kernel.org/lkml/20240411132853.GA26481@willie-the-truck/T/#m535f22236b46ba3a9e072fe8a91b0e89ec8597b7), but it was rejected.

I could be wrong, but since this isn’t in the Linux kernel, it seems taking advantage of this feature (and of course calling `prctl` to enable it) now rests elsewhere. I’d be surprised if OrbStack isn’t already taking advantage of this.

Any chance pre-patched kernels can be made available for Lima, for the typical, default “Docker VM” scenario?

**Comment by AkihiroSuda** [comment]
I don't think we have bandwidth to maintain our own kernel tree; maybe you should try negotiating with distro maintainers?
